test/e2e/tests/phishing-controller/phishing-detection.spec.js

```diff
@@ -297,4 +330,151 @@ describe('Phishing Detection', function () {
       },
     );
   });
+
+  describe('Phishing redirect protections', function () {
+    /**
+     * Status codes 305 (via Location header) and 306 (Set-Proxy) header do not
+     * automatically forward user agents to the new location, so we don't test
+     * these status codes. Status code 304 (Not Modified) is not a redirect
+     * status code.
+     *
+     * 201 _does_ require a Location header, but it also doesn't get
+     * automatically redirected by the user-agent, so we don't test it either.
+     *
+     * 200 is only included in order to test the `Refresh` header.
+     *
+     * @type {readonly [200 | 301 | 302 | 303 | 307 | 308]}
+     */
+    const redirectableStatusCodes = [200, 301, 302, 303, 307, 308];
+
+    const destination = 'https://metamask.github.io/test-dapp/';
+    const blocked = '127.0.0.1';
+
+    /**
+     * @type {Server | undefined}
+     */
+    let server;
+
+    /**
+     * @type {Driver | undefined}
+     */
+    let driver;
+
+    /**
+     * @type {Promise<void> | undefined}
+     */
+    let fixturePromise;
+
+    /**
+     *  Handle requests by setting the given header values and status code.
+     *
+     * @param {string} name - The name of the header to set.
+     * @param {string} value - The value of the header to set.
+     * @param {typeof redirectableStatusCodes[number]} code - The status code
+     * @returns {void}
+     */
+    function handleRequests(name, value, code) {
+      server.once('request', async function (_request, response) {
+        response.setHeader(name, value).writeHead(code).end(`
+        <!doctype html>
+        <html>
+          <head>
+            <meta http-equiv="Refresh" content="0;url="${destination}"/>
+            <title>Phishing test</title>
+
+            <script>
+              // this script should not run.
+              // it is meant to test for regressions in our redirect
+              // protection due to changes in either MetaMask or browsers.
+              document.location.href = "${destination}";
+              alert("trying to prevent phishing protection");
+              while(true){}
```

Comment:
Random ideas to try to work around our phishing protections. They don't affect our protections and are actually not necessary for our testing. I figured I could keep them here for regression testing. Thoughts?